src/Features/Core/Portable/MakeMethodSynchronous/AbstractMakeMethodSynchronousCodeFixProvider.cs

```diff
@@ -0,0 +1,110 @@
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var diagnostic = context.Diagnostics.First();
+
+            var token = diagnostic.Location.FindToken(cancellationToken);
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            for (var node = token.Parent; node != null; node = node.Parent)
+            {
+                if (IsMethodOrAnonymousFunction(node))
+                {
+                    context.RegisterCodeFix(new MyCodeAction(FeaturesResources.Make_method_synchronous,
+                        c => FixNodeAsync(context.Document, node, c)), diagnostic);
+                    return;
+                }
+            }
+        }
+
+        private const string AsyncSuffix = "Async";
+
+        private async Task<Solution> FixNodeAsync(Document document, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var methodSymbolOpt = semanticModel.GetDeclaredSymbol(node) as IMethodSymbol;
+
+            // If the method ends with 'Async' then remove that as part of the fix.
+            if (methodSymbolOpt?.MethodKind == MethodKind.Ordinary && 
+                methodSymbolOpt.Name.Length > AsyncSuffix.Length && 
+                methodSymbolOpt.Name.EndsWith(AsyncSuffix))
+            {
+                return await RenameThenRemoveAsyncTokenAsync(document, node, methodSymbolOpt, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                return await RemoveAsyncTokenAsync(document, methodSymbolOpt, node, cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        private async Task<Solution> RenameThenRemoveAsyncTokenAsync(Document document, SyntaxNode node, IMethodSymbol methodSymbol, CancellationToken cancellationToken)
+        {
+            var name = methodSymbol.Name;
+            var newName = name.Substring(0, name.Length - AsyncSuffix.Length);
+            var solution = document.Project.Solution;
+            var options = solution.Workspace.Options;
+
+            // Store the path to this node.  That way we can find it post rename.
+            var syntaxPath = new SyntaxPath(node);
+
+            // Rename the method to remove the 'Async' suffix, then remove the 'async' keyword.
+            var newSolution = await Renamer.RenameSymbolAsync(solution, methodSymbol, newName, options, cancellationToken).ConfigureAwait(false);
+            var newDocument = newSolution.GetDocument(document.Id);
+            var newRoot = await newDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            SyntaxNode newNode;
+            if (syntaxPath.TryResolve<SyntaxNode>(newRoot, out newNode))
```

Comment:
Nope. it's all indices based.  Indeed, it has a note in it saying "we *could* use the name in the future".